core/storage/suggestion/gae_models.py

```diff
@@ -0,0 +1,267 @@
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Models for Oppia suggestions."""
+
+from core.platform import models
+import feconf
+
+from google.appengine.ext import ndb
+
+(base_models,) = models.Registry.import_models([models.NAMES.base_model])
+
+# Constants defining types of entities to which suggestions can be created.
+ENTITY_TYPE_EXPLORATION = 'exploration'
+ENTITY_TYPE_QUESTION = 'question'
+ENTITY_TYPE_SKILL = 'skill'
+ENTITY_TYPE_TOPIC = 'topic'
+
+ENTITY_TYPE_CHOICES = [
+    ENTITY_TYPE_EXPLORATION,
+    ENTITY_TYPE_QUESTION,
+    ENTITY_TYPE_SKILL,
+    ENTITY_TYPE_TOPIC
+]
+
+# Constants defining the different possible statuses of a suggestion.
+STATUS_ACCEPTED = 'accepted'
+STATUS_IN_REVIEW = 'review'
+STATUS_INVALID = 'invalid'
+STATUS_REJECTED = 'rejected'
+
+STATUS_CHOICES = [
+    STATUS_ACCEPTED,
+    STATUS_IN_REVIEW,
+    STATUS_INVALID,
+    STATUS_REJECTED
+]
+
+# Constants defining various suggestion types.
+SUGGESTION_EDIT_STATE_CONTENT = 'edit_exploration_state_content'
+
+SUGGESTION_TYPE_CHOICES = [
+    SUGGESTION_EDIT_STATE_CONTENT
+]
+
+# Defines what is the minimum role required to review suggestions
+# of a particular type.
+SUGGESTION_MINIMUM_ROLE_FOR_REVIEW = {
+    SUGGESTION_EDIT_STATE_CONTENT: feconf.ROLE_ID_EXPLORATION_EDITOR
+}
+
+# Constants defining various contribution types.
+CONTRIBUTION_TYPE_CONTENT = 'content'
+CONTRIBUTION_TYPE_TRANSLATION = 'translation'
+
+CONTRIBUTION_TYPE_CHOICES = [
+    CONTRIBUTION_TYPE_CONTENT,
+    CONTRIBUTION_TYPE_TRANSLATION
+]
+
+#The delimiter to be used in the suggestion ID.
+SUGGESTION_ID_DELIMITER = '.'
+
+
+class SuggestionParams(ndb.Model):
```

Comment:
SuggestionMetadata is probably a better name, I think. That's what this is -- the data "around" the main content of the suggestion.

Reply:
Made these first class parameters as discussed.